cmd/outline-ss-server/config.go

```diff
@@ -22,71 +22,128 @@ import (
 )
 
 type ServiceConfig struct {
-	Listeners []ListenerConfig
-	Keys      []KeyConfig
-	Dialer    DialerConfig
+	Listeners []ListenerConfig `yaml:"listeners"`
+	Keys      []KeyConfig      `yaml:"keys"`
+	Dialer    DialerConfig     `yaml:"dialer"`
 }
 
 type ListenerType string
 
-const listenerTypeTCP ListenerType = "tcp"
+const (
+	listenerTypeTCP ListenerType = "tcp"
 
-const listenerTypeUDP ListenerType = "udp"
+	listenerTypeUDP             ListenerType = "udp"
+	listenerTypeWebsocketStream ListenerType = "websocket-stream"
+	listenerTypeWebsocketPacket ListenerType = "websocket-packet"
+)
+
+type WebServerConfig struct {
+	ID        string   `yaml:"id"`
+	Listeners []string `yaml:"listen"`
+}
 
 type ListenerConfig struct {
```

Comment:
Can you add comments here?

Also, this needs to be redesigned. Each type needs different parameters, but we can't simply merge them all in one struct. They each need separate definitions.